src/Kiota.Builder/Plugins/PluginsGenerationService.cs

```diff
@@ -184,79 +192,120 @@ descriptionExtension is OpenApiDescriptionForModelExtension extension &&
             privacyUrl = privacy.Privacy;
 
         return new OpenApiManifestInfo(descriptionForModel, legalUrl, logoUrl, privacyUrl, contactEmail);
-
     }
+
     private const string DefaultContactName = "publisher-name";
     private const string DefaultContactEmail = "publisher-email@example.com";
-    private sealed record OpenApiManifestInfo(string? DescriptionForModel = null, string? LegalUrl = null, string? LogoUrl = null, string? PrivacyUrl = null, string ContactEmail = DefaultContactEmail);
-    private static (OpenApiRuntime[], Function[]) GetRuntimesAndFunctionsFromTree(OpenApiUrlTreeNode currentNode, string openApiDocumentPath)
+
+    private sealed record OpenApiManifestInfo(
+        string? DescriptionForModel = null,
+        string? LegalUrl = null,
+        string? LogoUrl = null,
+        string? PrivacyUrl = null,
+        string ContactEmail = DefaultContactEmail);
+
+    private static (OpenApiRuntime[], Function[]) GetRuntimesAndFunctionsFromTree(OpenApiDocument document, PluginAuthConfiguration? authInformation, OpenApiUrlTreeNode currentNode,
+        string openApiDocumentPath)
     {
         var runtimes = new List<OpenApiRuntime>();
         var functions = new List<Function>();
+        var configAuth = authInformation?.ToPluginManifestAuth();
         if (currentNode.PathItems.TryGetValue(Constants.DefaultOpenApiLabel, out var pathItem))
         {
             foreach (var operation in pathItem.Operations.Values.Where(static x => !string.IsNullOrEmpty(x.OperationId)))
             {
                 runtimes.Add(new OpenApiRuntime
                 {
-                    Auth = new AnonymousAuth(),
-                    Spec = new OpenApiRuntimeSpec()
-                    {
-                        Url = openApiDocumentPath,
-                    },
+                    // Configuration overrides document information
+                    Auth = configAuth ?? GetAuth(operation.Security ?? document.SecurityRequirements),
+                    Spec = new OpenApiRuntimeSpec { Url = openApiDocumentPath, },
                     RunForFunctions = [operation.OperationId]
                 });
                 functions.Add(new Function
                 {
                     Name = operation.OperationId,
                     Description =
-                        operation.Summary.CleanupXMLString() is string summary && !string.IsNullOrEmpty(summary)
+                        operation.Summary.CleanupXMLString() is { } summary && !string.IsNullOrEmpty(summary)
                             ? summary
                             : operation.Description.CleanupXMLString(),
                     States = GetStatesFromOperation(operation),
                 });
+                continue;
```

Comment:
Actually, is this continue expected without a condition? should we change to a first or default instead if we always want a single runtime/function?

Reply:
The continue is meant to visually separate the loop from the local function just below it.

Reply:
Ah, continue, not break. 
Could we instead move the local function outside of the loop? Or make it a method, for clarity.